File: Evergrande/src/gestorAplicacion/otros/UnidadResidencial.java
package gestorAplicacion.otros;

import java.io.Serializable;
import java.util.ArrayList;

/**
 * @Autores:
 * David Escobar Ruiz
 * Julián Orozco Vanegas
 * Juan Nicolas Piedrahita Salas
 * 
 * @Descripcion
 * Esta clase representa las unidades residenciales asociadas a cada inmueble
 *
 */

public abstract class UnidadResidencial implements Ciudad, Serializable{
	private static final long serialVersionUID = 1L;
	private int idUnidadResidencial;
	private String nombre;
	private String barrio;
	private static ArrayList<UnidadResidencial> unidades = new ArrayList<UnidadResidencial>();
	
    public UnidadResidencial(int idUnidadResidencial, String nombre, String barrio) {
        this.idUnidadResidencial = idUnidadResidencial;
        this.nombre = nombre;
        this.barrio = barrio;
    }
    
    public int getIdUnidadResidencial() {
        return this.idUnidadResidencial;
    }

    public void setIdUnidadResidencial(int idUnidadResidencial) {
        this.idUnidadResidencial = idUnidadResidencial;
    }

    public String getNombre() {
        return this.nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public String getBarrio() {
        return this.barrio;
    }

    public void setBarrio(String barrio) {
        this.barrio = barrio;
    }
    
    public static ArrayList<UnidadResidencial> getUnidades() {
        return UnidadResidencial.unidades;
    }

    public static void setUnidades(ArrayList<UnidadResidencial> unidades) {
        UnidadResidencial.unidades = unidades;
    }

}

File: Python/src/gestorAplicacion/otros/UnidadResidencial.py
'''
Autores: 
Julián Orozco Vanegas
Juan Nicolas Piedrahita Salas
David Escobar Ruiz

Esta clase simula el comportamiento de una unidad residencial y sus caracteristicas generales
'''

from .Ciudad import Ciudad

class UnidadResidencial(Ciudad):
  UNIDADES = []
  def __init__(self, idUnidadResidencial, nombre, barrio):
    self.idUnidadResidencial = idUnidadResidencial
    self.nombre = nombre
    self.barrio = barrio
    UnidadResidencial.UNIDADES.append(self)
  
  @classmethod
  def getUnidades(cls):
    return UnidadResidencial.UNIDADES
  
  def getIdUnidadResidencial(self):
    return self.idUnidadResidencial
  
  def getNombre(self):
    return self.nombre
  
  def getBarrio(self):
    return self.barrio
  